app/assets/javascripts/test/libs/nml.spec.js

```diff
@@ -0,0 +1,185 @@
+// @flow
+
+import test from "ava";
+import _ from "lodash";
+import mock from "mock-require";
+import { defaultState } from "oxalis/store";
+import update from "immutability-helper";
+import type { NodeType } from "oxalis/store";
+
+const TIMESTAMP = 123456789;
+const DateMock = {
+  now: () => TIMESTAMP,
+};
+mock("libs/date", DateMock);
+const { serializeToNml, getNmlName, parseNml } = mock.reRequire("oxalis/model/helpers/nml_helpers");
+const SkeletonTracingReducer = mock.reRequire("oxalis/model/reducers/skeletontracing_reducer")
+  .default;
+const SkeletonTracingActions = mock.reRequire("oxalis/model/actions/skeletontracing_actions");
+
+const createDummyNode = (id: number): NodeType => ({
+  bitDepth: 8,
+  id,
+  position: [0, 0, 0],
+  radius: 10,
+  resolution: 10,
+  rotation: [0, 0, 0],
+  timestamp: id,
+  viewport: 1,
+  interpolation: true,
+});
+
+const tracing = {
+  type: "skeleton",
+  trees: {
+    "1": {
+      treeId: 1,
+      name: "TestTree-0",
+      nodes: {
+        "0": createDummyNode(0),
+        "1": createDummyNode(1),
+        "2": createDummyNode(2),
+        "7": createDummyNode(7),
+      },
+      timestamp: TIMESTAMP,
+      branchPoints: [{ nodeId: 1, timestamp: 0 }, { nodeId: 7, timestamp: 0 }],
+      edges: [{ source: 0, target: 1 }, { source: 2, target: 1 }, { source: 1, target: 7 }],
+      comments: [{ content: "comment", nodeId: 0 }],
+      color: [23, 23, 23],
+      isVisible: true,
+    },
+    "2": {
+      treeId: 2,
+      name: "TestTree-1",
+      nodes: {
+        "4": createDummyNode(4),
+        "5": createDummyNode(5),
+        "6": createDummyNode(6),
+      },
+      timestamp: TIMESTAMP,
+      branchPoints: [],
+      edges: [{ source: 4, target: 5 }, { source: 5, target: 6 }],
+      comments: [],
+      color: [30, 30, 30],
+      isVisible: true,
+    },
+  },
+  tracingType: "Explorational",
+  name: "",
+  activeTreeId: 1,
+  activeNodeId: 1,
+  annotationId: "annotationId",
+  cachedMaxNodeId: 7,
+  restrictions: {
+    branchPointsAllowed: true,
+    allowUpdate: true,
+    allowFinish: true,
+    allowAccess: true,
+    allowDownload: true,
+  },
+};
+
+const initialState = _.extend({}, defaultState, {
+  tracing,
+  activeUser: { firstName: "SCM", lastName: "Boy" },
+});
+
+async function throwsAsync(t, fn) {
+  try {
+    await fn.call();
+    t.fail(`Test did not throw, calling the following function: ${fn.toString()}`);
+  } catch (e) {
+    t.true(true);
```

Comment:
If we had a dedicated `NmlParseError` class, we could check whether an expected parse error was thrown or something else went wrong.

Reply:
That sounds like a very good idea :)